src/control/SwerveController.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "../kinematics/DiffDriveKinematics.h"
+#include "../kinematics/SwerveDriveKinematics.h"
+
+#include <map>
+
+namespace control {
+enum class DriveMode {
+	Normal,
+	TurnInPlace,
+	Crab,
+};
+
+static const std::map<DriveMode, std::string> driveModeStrings = {
+	{DriveMode::Normal, "Normal"},
+	{DriveMode::TurnInPlace, "TurnInPlace"},
+	{DriveMode::Crab, "Crab"}};
+
+// Represents the rotations of all four steer motors in in millidegrees.
+struct swerve_rots_t {
+	int lfRot;
+	int rfRot;
+	int lbRot;
+	int rbRot;
+};
+
+// The PWM commands to all four steer motors.
+struct swerve_commands_t {
+	double lfPWM;
+	double rfPWM;
+	double lbPWM;
+	double rbPWM;
```

Comment:
Let's use `power` instead of `pwm`, since the exact method of setting the power is an implementation detail. For example, BLDC motors don't even use PWM.